src/udp.c

```diff
@@ -233,6 +258,49 @@ ssize_t trudpUdpRecvfrom(int fd, uint8_t* buffer, size_t buffer_size,
     ssize_t recvlen = recvfrom(fd, buffer, buffer_size, flags,
             (__SOCKADDR_ARG)remaddr, addr_length);
 
+    if (recvlen == -1) {
+#if defined(TEONET_OS_WINDOWS)
+        int recv_errno = WSAGetLastError();
+#else
+        int recv_errno = errno;
+#endif
+
+#if defined(TEONET_OS_WINDOWS)
+        if (recv_errno != WSAEWOULDBLOCK) {
+#else
+        // EWOULDBLOCK may be not defined or may have same value as EAGAIN.
+#if defined(EWOULDBLOCK) && EWOULDBLOCK != EAGAIN
+        if (recv_errno != EAGAIN && recv_errno != EWOULDBLOCK) {
+#else
+        if (recv_errno != EAGAIN) {
+#endif
+#endif
+            // TODO: Use thread safe error formatting function.
+            // TODO: On Windows use correct error formatting function.
+            LTRACK_E("TrUdp",
+                "Receiving data using recvfrom() failed with error %" PRId32
+                ": %s", recv_errno, strerror(recv_errno));
+        }
+    } else if (recvlen == 0) {
+        LTRACK_E("TrUdp", "Receiving data using recvfrom() returned 0 (connection closed).");
+    } else {
+        if (trudpOpt_DBG_dumpUdpData) {
+            char hexdump[32];
+            if (buffer != NULL && buffer_size > 0) {
+                dump_bytes(hexdump, sizeof(hexdump), buffer, recvlen);
+            } else {
+                strcpy(hexdump, "(null)");
+            }
+
+            dump_bytes(hexdump, sizeof(hexdump), buffer, recvlen);
+            LTRACK("TrUdp",
+                   "Received %u bytes using recvfrom() starting with %s",
+                   (uint32_t)recvlen, hexdump);
+        }
+
+        _trudpCallUdpDataReceivedCallback(recvlen);
```

Comment:
I think we need define (#ifdef) here to exclude this function call in server (in teonet and teocli) (this call added for client only).